### ISSUE TITLE
Bad color in graph for toner translated label

### ISSUE THREAD

**Issue by tuxi56** [librenms/librenms#4404]
Hi

I have a color issue in graphs for my Ricoh printer : black and yellow toner aren't using the proper color : 

<img width="658" alt="capture d ecran 2016-09-12 a 04 04 19" src="https://cloud.githubusercontent.com/assets/22138991/18422668/099fc056-789e-11e6-9b43-3b80ea54a213.png">

As you can see the label is the translated color name ("noir" for black and "jaune" for yellow). I guess this is not a problem for cyan and magenta as long as the translated name is the same as in English ?

There is also a strange "?" instead of "usagé" (not an actual toner color, but the used one) but it's not a big problem for that one. 

Is there a way to force/change the color used by RRD for the graphs ?

I have no issue with other printer manufacturers we use (Brother and Canon), as long as the toner name is using the English color names for those ones. 

Here is my ./validate.php : 

```
Version info:
Commit SHA: d1ae83a378a6016a6619f6f13d49209334baf2c0
DB Schema: 118
PHP: 7.0.10-1~dotdeb+8.1
MySQL: 5.5.50-0+deb8u1
RRDTool: 1.4.8
SNMP: NET-SNMP version: 5.7.2.1
```

I also added the `$config['enable_printers'] = true;` in my `config.php` (but it worked the same without this directive). 

Here's the discovery.php output  : http://pastebin.com/0SSQuLXj
Here's the pooler.php output : http://pastebin.com/h7GHTAX5

Thank you :)

**Comment by tuxi56** [comment]
After manually editing the `toner_descr` field in the database (table `toner`) with the English values, now the colors are matching (both on the Overview and on the Toner graph) : 

<img width="663" alt="capture d ecran 2016-09-12 a 05 28 04" src="https://cloud.githubusercontent.com/assets/22138991/18423892/be11658e-78a9-11e6-92d9-85c0e3f0117e.png">

<img width="219" alt="capture d ecran 2016-09-12 a 05 28 15" src="https://cloud.githubusercontent.com/assets/22138991/18423893/be121d80-78a9-11e6-9658-ce751b6be83f.png">

**Comment by tuxi56** [comment]
Unfortunately, after some update, the text in database changed to the original value, thus the graph color also.

**Comment by tuxi56** [comment]
Unfortunately, after some update, the text in database changed to the original value, also did the graph color. 

I wrote a temporary fix, running a script via the crontab to re-update the values.

**Comment by laf** [comment]
It will always be overwritten. We will need to put something in place but this is an interesting challenge as we can't really provide a lookup for all languages.

**Comment by tuxi56** [comment]
Hi @laf

For the moment I consider this issue fixed with this workaround, I guess i18n is a big challenge :) 

Wouldn't it be better to catch the toner color from the oid instead of the text issued by the device?

**Comment by crcro** [comment]
i think that there is something in ricoh mibs regarding multi-language oids ... i'll take a look

**Comment by laf** [comment]
Fixed in #5410
